### PR TITLE
add history query.

### LINKED ISSUE #3195: 可以提供一下某个配置的变更记录查询。现在只能一个一个翻，很麻烦
rt

### PR DESCRIPTION
**Pull request by Accelerater**
## What's the purpose of this PR

添加了通过配置的key对变更历史的查询

## Which issue(s) this PR fixes:
Fixes #3195

## Brief changelog

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Read the [Contributing Guide](https://github.com/ctripcorp/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit tests to verify the code.
- [x] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [x] Update the [`CHANGES` log](https://github.com/ctripcorp/apollo/blob/master/CHANGES.md).

### CODE REVIEW

File: apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/CommitService.java [hunk 1]
```diff
@@ -51,6 +51,11 @@ public Commit save(Commit commit){
             appId, clusterName, namespaceName, lastModifiedTime, page);
   }
 
+  public List<Commit> findByKey(String appId, String clusterName, String namespaceName, String key,Pageable page){
+    String queryKey = "\"key\":\""+ key +"\"";
+    return commitRepository.findByAppIdAndClusterNameAndNamespaceNameAndChangeSetsLikeOrderByIdDesc(appId, clusterName, namespaceName, "%"+ queryKey + "%", page);
```
Comment (nobodyiam) [inline apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/CommitService.java:56]: What is the performance impact of this query?

Comment (Accelerater) [reply apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/CommitService.java:56]: 本地测试1w条左右数据量几乎没有影响。

Comment (Accelerater) [reply apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/CommitService.java:56]: @nobodyiam 宋哥 这么实现你觉得还有什么问题么？

Comment (JaredTan95) [reply apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/CommitService.java:56]: filtered by `appid`, `clustername`, `namespace` etc，1w+ record has covered almost cases. 

LGTM

**Review by JaredTan95 — APPROVED** [review]

**Pull request merged**